### PR TITLE
Add lowball broadcast support with API key for all transaction types

### LINKED ISSUE #382: Use lowball for non cooperative refund

### LINKED ISSUE #423: Send Payment: Use lowball fees when sending onchain (via magic routing hint BIP21)
When `send_payment` uses the lowball feerate for sending via swaps and the normal feerate (10x higher) for sending onchain.

We should use the lowball feerate in this situation too.

One way to do it is:
- include the invoice in the broadcast API request
- the Swapper can use the Boltz API to check if the onchain destination address is in the MRH BIP21 in the invoice
  https://api.boltz.exchange/swagger#/Reverse/get_swap_reverse__invoice__bip21
  `/swap/reverse/{invoice}/bip21`
- the Swapper can check the invoice route hints and use Boltz API to verify they are connected to a swap
- if all checks succeed, broadcast tx with lowball feerate

### PR DESCRIPTION
**Pull request by hydra-yse**
Closes #423, closes #382.

### CODE REVIEW

File: lib/core/src/model.rs [hunk 1]
```diff
@@ -48,6 +48,8 @@ pub struct Config {
     /// Maximum amount in satoshi to accept zero-conf payments with
     /// Defaults to [crate::receive_swap::DEFAULT_ZERO_CONF_MAX_SAT]
     pub zero_conf_max_amount_sat: Option<u64>,
+    /// The Breez API key used for making requests to
```
Comment (ok300) [inline lib/core/src/model.rs:51]: The comment is incomplete.

**Review by ok300 — COMMENTED** [review]
I think you missed using lowball for MRH txs. Both prepare and `pay_liquid` reference the standard fee rate internally.

File: lib/core/src/model.rs (hunk 1 above)
Comment (hydra-yse) [reply lib/core/src/model.rs:51]: Weird, I thought I pushed the changes

**Review by ok300 — COMMENTED** [review]
Should `pay_liquid` also be adjusted?

Internally it uses `utils::derive_fee_rate_sats_per_kvb` to get the fee rate, which itself uses the standard fee rate.

**Review by roeierez — COMMENTED** [review]
Looks good! dropped one question.

File: lib/core/src/receive_swap.rs [hunk 2]
```diff
@@ -243,24 +246,46 @@ impl ReceiveSwapStateHandler {
...
+            .broadcast(&claim_tx, Some(&swap.id))
+            .await;
+        let claim_tx_id = match broadcast_response {
+            Ok(tx_id) => tx_id.to_hex(),
+            Err(err) => {
+                debug!(
+                    "Could not broadcast claim tx via chain service for swap {swap_id}: {err:?}"
+                );
+                let claim_tx_hex = claim_tx.serialize().to_lower_hex_string();
+                self.swapper
```
[the remark this answers is not in the document]
Comment (hydra-yse) [reply lib/core/src/receive_swap.rs:277]: Yes, originally this was a broadcast via Boltz so we can broadcast for sure via their endpoint. Meanwhile for the others it has always been with the Liquid chain service, and I'm unsure whether or not Boltz will reject it.

**Review by roeierez — APPROVED** [review]
LGTM

**Review by ok300 — APPROVED** [review]

**Comment by hydra-yse** [comment]
@ok300 Addressed [the comment](https://github.com/breez/breez-sdk-liquid/pull/407#discussion_r1765183393) from 407 in 7dd0932. Let me know if you agree :+1:

**Review by ok300 — APPROVED** [review]
Looks good!

**Review by dangeross — COMMENTED** [review]
Just some notes about docs, we need to update the CLI docs for setting the BREEZ_API_KEY in the env and the SDK docs really need to highlight in Getting Started and Send Payment setting the key and the limitations of not setting it.

File: cli/src/main.rs [hunk 3]
```diff
@@ -73,6 +73,8 @@ async fn main() -> Result<()> {
     let network = args.network.unwrap_or(LiquidNetwork::Testnet);
     let mut config = LiquidSdk::default_config(network);
     config.working_dir = data_dir_str;
+    config.breez_api_key = std::env::var_os("BREEZ_API_KEY")
+        .map(|var| var.into_string().expect("Invalid API key provided"));
```
[the remark this answers is not in the document]
Comment (hydra-yse) [reply cli/src/main.rs:77]: Addressed in breez/breez-sdk-liquid-docs#36

**Pull request merged**